tests/system/plugin.test.ts

```diff
@@ -313,3 +310,29 @@ test('inspect on mvnw is successful with targetFile', async (t) => {
   delete result.plugin.meta;
   t.same(result, expected, 'should return expected result');
 });
+
+test('inspect on mvnw is successful with targetFile in sub folder', async (t) => {
+  const result = await plugin.inspect(
+    path.join(fixturesPath, 'maven-with-mvnw-sub'),
+    path.join('sub', 'pom.xml'),
+  );
+  if (legacyPlugin.isMultiResult(result)) {
+    return t.fail('expected single inspect result');
+  }
+  const expected = await readFixtureJSON(
+    'maven-with-mvnw-sub',
+    'expected.json',
+  );
+  // result.metadata depends on platform, so no fixture can be provided
+  t.ok(
+    result!.plugin!.meta!.versionBuildInfo!.metaBuildVersion!.javaVersion,
```

Comment:
For lint warning purposes, I would replace all `!` with `?`.
So the result if smth not exists will be `undefined`